pallet-chainlink-feed/src/lib.rs

```diff
@@ -0,0 +1,1345 @@
+use frame_support::traits::{Currency, ExistenceRequirement, Get, ReservableCurrency};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::{DispatchError, DispatchResult, DispatchResultWithPostInfo, HasCompact},
+	ensure, Parameter, RuntimeDebug,
+};
+use frame_system::ensure_signed;
+use sp_arithmetic::traits::BaseArithmetic;
+use sp_runtime::{
+	traits::{AccountIdConversion, CheckedAdd, CheckedSub, Member, One, Saturating, Zero},
+	ModuleId,
+};
+use sp_std::convert::{TryFrom, TryInto};
+
+/// Execute the supplied function in a new storage transaction.
+///
+/// All changes to storage performed by the supplied function are discarded if
+/// the returned outcome is `Result::Err`.
+///
+/// Transactions can be nested to any depth. Commits happen to the parent
+/// transaction.
+// TODO: remove after move to Substrate v3 (once the semantics of #[transactional] work as intended)
+pub(crate) fn with_transaction_result<R, E>(f: impl FnOnce() -> Result<R, E>) -> Result<R, E> {
+	with_transaction(|| {
+		let res = f();
+		if res.is_ok() {
+			TransactionOutcome::Commit(res)
+		} else {
+			TransactionOutcome::Rollback(res)
+		}
+	})
+}
+
+/// Determine the median of a slice of values.
+pub(crate) fn median<T: Copy + BaseArithmetic>(numbers: &mut [T]) -> T {
+	numbers.sort_unstable();
+
+	let mid = numbers.len() / 2;
+	if numbers.len() % 2 == 0 {
+		numbers[mid - 1].saturating_add(numbers[mid]) / 2.into()
+	} else {
+		numbers[mid]
+	}
+}
+
+pub type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+
+pub trait Trait: frame_system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+	/// Type for feed indexing.
+	type FeedId: Member + Parameter + Default + Copy + HasCompact + BaseArithmetic;
+
+	/// Type for round indexing.
+	type RoundId: Member + Parameter + Default + Copy + HasCompact + BaseArithmetic + Into<u64>;
+
+	/// Oracle feed values.
+	type Value: Member + Parameter + Default + Copy + HasCompact + PartialEq + BaseArithmetic;
+
+	/// Interface used for balance transfers.
+	type Currency: ReservableCurrency<Self::AccountId>;
+
+	/// The module id used to determine the account for storing the funds used to pay the oracles.
+	type ModuleId: Get<ModuleId>;
+
+	/// The minimum amount of funds that need to be present in the fund account.
+	type MinimumReserve: Get<BalanceOf<Self>>;
+
+	/// Maximum allowed string length.
+	type StringLimit: Get<u32>;
+
+	/// Maximum number of oracles per feed.
+	type OracleCountLimit: Get<u32>;
+
+	/// Maximum number of feeds.
+	type FeedLimit: Get<Self::FeedId>;
+
+	/// Number of rounds to keep around per feed.
+	type PruningWindow: Get<Self::RoundId>;
+
+	// type WeightInfo: WeightInfo;
+}
+
+/// The configuration for an oracle feed.
+#[derive(Clone, Encode, Decode, Default, Eq, PartialEq, RuntimeDebug)]
+pub struct FeedConfig<
```

Comment:
as discussed elsewhere you can actually have `FeedConfig<T: Config>` to make life simpler, but honestly an alias like `FeedConfigOf` is also okay. If you use `T: Config` you need to use some special derive macros, or implement things manually.